monitoring/monitoring/monitoring.py

```diff
@@ -0,0 +1,219 @@
+import datetime
+import calendar
+import asyncio
+from aiohttp import web
+import aiohttp_session
+import logging
+from collections import defaultdict
+
+from hailtop import aiogoogle
+from hailtop.aiogoogle import BigQueryClient
+from hailtop.config import get_deploy_config
+from hailtop.hail_logging import AccessLogger
+from hailtop.tls import get_in_cluster_server_ssl_context
+from hailtop.utils import run_if_changed_idempotent, retry_long_running, time_msecs
+from gear import (Database, setup_aiohttp_session,
+                  web_authenticated_developers_only, transaction)
+from web_common import (setup_aiohttp_jinja2, setup_common_static_routes,
+                        render_template, set_message)
+
+log = logging.getLogger('monitoring')
+
+routes = web.RouteTableDef()
+
+deploy_config = get_deploy_config()
+
+
+def get_previous_month(dt):
+    if dt.month == 1:
+        return datetime.datetime(dt.year - 1, 12, 1)
+    return datetime.datetime(dt.year, dt.month - 1, 1)
+
+
+def get_last_day_month(dt):
+    _, last_day = calendar.monthrange(dt.year, dt.month)
+    return datetime.datetime(dt.year, dt.month, last_day)
+
+
+def format_data(records):
+    cost_by_service = defaultdict(lambda: 0)
+    compute_cost_breakdown = defaultdict(lambda: 0)
+    cost_by_sku_label = []
+
+    for record in records:
+        cost_by_sku_label.append(record)
+
+        cost_by_service[record['service_description']] += record['cost']
+
+        if record['service_description'] == 'Compute Engine':
+            compute_cost_breakdown[record['source']] += record['cost']
+
+    cost_by_service = sorted([{'service': k, 'cost': v} for k, v in cost_by_service.items()], key=lambda x: x['cost'], reverse=True)
+    compute_cost_breakdown = sorted([{'source': k, 'cost': v} for k, v in compute_cost_breakdown.items()], key=lambda x: x['cost'], reverse=True)
```

Comment:
These lines are a mouthful! Can we try newlines between the arguments? I think that might make it easier to see what's happening.